db/compaction_job.cc

```diff
@@ -720,13 +720,13 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
       kUseZstdTrainer ? cfd->ioptions()->compression_opts.zstd_max_train_bytes
                       : cfd->ioptions()->compression_opts.max_dict_bytes;
   const int kSampleLenShift = 6;  // 2^6 = 64-byte samples
-  std::set<size_t> sample_begin_offsets;
+  std::set<uint64_t> sample_begin_offsets;
   if (bottommost_level_ && kSampleBytes > 0) {
     const size_t kMaxSamples = kSampleBytes >> kSampleLenShift;
-    const size_t kOutFileLen = mutable_cf_options->MaxFileSizeForLevel(
+    const uint64_t kOutFileLen = mutable_cf_options->MaxFileSizeForLevel(
```

Comment:
would it work to cast MaxFileSizeForLevel output to a size_t instead? might be simpler